### DISCUSSION TITLE
Is it possible to do sth when test complets even the test is failed?

### DISCUSSION THREAD

**Discussion by Juliehzl** [xunit/xunit#2776] [Question] [answered]
I want to do some actions when test is finished even it is failed. I also want to know the test result failed or succeed.

Is it possible to use BeforeAfterTestAttribute? But I only see there is method info instead of test result.

**Comment by bradwilson** [comment, marked as answer]
There is no way to do this in v2.

In v3, we've added [`TestContext`](https://github.com/xunit/xunit/blob/main/src/xunit.v3.core/TestContext.cs) which offers `TestState` which provides access to the result, exception information, and failure cause. We have not yet shipped the first prerelease build of v3.
